BWIndex.py

```diff
@@ -0,0 +1,239 @@
+from graphviz import Digraph
+import sys, argparse
+
+
+class BWIndex():
+	def __init__(self, genome=None, genomefile=None):
+		if genome is not None:
+			self.Text = genome
+		elif genomefile is not None:
+			self.loadGenomeFromFile(genomefile)
+		else:
+			print("You must supply a genome file or a genome string.")
+			sys.exit()
+		if self.Text[-1] != '$':
+			print("Genome must be terminated by $")
+			sys.exit()
+		self.buildTree()
+		self.Solve()
+		self.Count = dict()
+		self.letters = {i for i in self.BWT}
+		self.alphabet = [i for i in self.letters]
+		self.alphabet.sort()
+		self.computeCountArray()
+
+		self.FirstOccurrence = dict()
+		freq = dict()
+		for i in self.BWT:
+			freq[i] = freq.get(i, 0) + 1

```

Comment:
For this pattern, you could use a `collections.Counter` or a `defaultdict(int)` to at least set up the default value of 0.